### PR TITLE
libarchive: disable linking with pcre2

### PR DESCRIPTION
**Pull request by Matthiasfranck**
Maintainer: @neheb 
Compile tested: (put here arch, model, OpenWrt version)
Run tested: (put here arch, model, OpenWrt version, tests done)

Description:
Looks like libarchive v3.7.3 adds support for pcre2 (https://github.com/libarchive/libarchive/commit/d1231a7ea7df4db6ddba73f06df67dbab0174a5c).

Because of this, the following build error is observed (in the case pcre2 is compiled before libarchive): Package libarchive is missing dependencies for the following libraries: libpcre2-8.so.0
libpcre2-posix.so.3

So let's disable linking with pcre2, just like linking with pcre was disabled before.

### CODE REVIEW

File: libs/libarchive/Makefile [hunk 1]
```diff
@@ -78,6 +78,7 @@ CMAKE_OPTIONS += \
 	-DENABLE_ZSTD=OFF \
 	-DENABLE_LIBXML2=OFF \
 	-DENABLE_PCREPOSIX=OFF \
```
Comment (BKPepe) [inline libs/libarchive/Makefile:80]: We can drop this, right?

Comment (Matthiasfranck) [reply libs/libarchive/Makefile:80]: why? Looks like it is still used: https://github.com/libarchive/libarchive/blob/v3.7.4/CMakeLists.txt#L225C1-L225C86

Comment (neheb) [reply libs/libarchive/Makefile:80]: libpcre is gone I think

Comment (Matthiasfranck) [reply libs/libarchive/Makefile:80]: let me know if I need to adapt this or not :)

**Pull request merged**